File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
  const init = Array.from({length:n},()=>1)
  const ch = init.map((a, i)=>  {
    if(reserve.includes(i+1)){
      if(lost.includes(i+1)){
        return a
      } 
      return a+=1
    } 
    if(lost.includes(i+1)){
      if(reserve.includes(i+1)){
        return a
      }
      return a-=1
    }
      return a
  })

  for(let i = 0; i < n; i++){
    if(!ch[i]){
      if(ch[i-1]>1){
        ch[i-1]--
        ch[i]++
      } else if(ch[i+1]>1){
        ch[i+1]--
        ch[i]++
      }
    }
  }
  return ch.filter(a=>a).length
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] students = new int[n+1];
    
        Arrays.fill(students, 1);
        Arrays.sort(reserve);
        
        for(int l : lost){
            students[l] -= 1;
        }
        
        for(int r : reserve){
            students[r] += 1;
        }
        
        for (int r : reserve) {
            int cur = r;
            int prev = r - 1;
            int next = r + 1;
           
            if(students[cur] < 2) continue;
            
            if (prev > 0 && students[prev] == 0) {
                students[prev] += 1; 
                students[cur] -= 1; 
            } else if (next <= n && students[next] == 0) {
                students[next] += 1; 
                students[cur] -= 1; 
            }   
        }
        
        return count(students);
    }
    
    
    public static int count(int[] array) {
        int count = 0;
        for (int i = 1; i < array.length; i++) {
            if (array[i] > 0) {
                count++;
            }
        }
        return count;
    }
}